### PR TITLE
PIL-1089: onboarding ens name fixes

### PR DESCRIPTION
**Pull request by poocart**
https://linear.app/pillarproject/issue/PIL-1089/user-tried-importing-choose-a-username-screen-appears

Includes:
- Added new Etherspot SDK method `validateENSName` that checks provided username for being invalid/blacklisted/taken.
- Added missing reserved ENS name check.
- Fixed previously unnoticed change of incorrect Etherspot SDK method rename.

### CODE REVIEW

File: src/services/etherspot.js [hunk 1]
```diff
@@ -292,7 +292,7 @@ export class EtherspotService {
 
   reserveEnsName(username: string): Promise<?ENSNode> {
     const fullEnsName = getEnsName(username);
-    return this.sdk.getAccountTokenListTokens({ name: fullEnsName }).catch((error) => {
+    return this.sdk.reserveENSName({ name: fullEnsName }).catch((error) => {
```
Comment (poocart) [inline src/services/etherspot.js:295]: This is silly :)

Comment (IAmKio) [reply src/services/etherspot.js:295]: Copy pasta?

Comment (mdjastrzebski) [reply src/services/etherspot.js:295]: 😂

File: src/services/onboarding.js [hunk 2]
```diff
@@ -81,6 +99,10 @@ export const isUsernameTaken = async (username: string): Promise<boolean> => {
 
   try {
     const ensName = getEnsName(username);
+
+    const isValid = await etherspotService.isValidEnsName(ensName);
+    if (!isValid) return true; // invalid/blacklisted/taken
```
Comment (mdjastrzebski) [inline src/services/onboarding.js:104]:
```suggestion
    if (!isValid) return true; // invalid/blacklisted/taken/error
```

**Review by IAmKio — APPROVED** [review]
LGTM

File: src/services/etherspot.js [hunk 3]
```diff
@@ -305,6 +305,14 @@ export class EtherspotService {
     });
   }
 
+  isValidEnsName(name: string): Promise<boolean> | boolean {
+    try {
+      return this.sdk.validateENSName({ name });
+    } catch (error) {
+      return false; // no need to report
```
Comment (mdjastrzebski) [inline src/services/etherspot.js:312]: I would consider throwing here, as failed network call will be reported as failure.

**Review by mdjastrzebski — APPROVED** [review]
Looks goods,

I would consider indicating error when isValidEnsName fails rather than silently treating it the same as reserved.

File: src/services/etherspot.js (hunk 1 above)
Comment (poocart) [reply src/services/etherspot.js:295]: I noticed that during last week this is second time this happens 👀  hot weather? rest? 🌴

**Pull request merged**